### ISSUE TITLE
no method for reporting installed state

### ISSUE THREAD

**Issue by andrewelizondo** [opscode-cookbooks/chef-server-ingredient#4]
```
Generated at 2015-01-16 21:57:48 +0000
NoMethodError: undefined method `installed' for Chef::Resource::ChefServerIngredient
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.16.2/lib/chef/resource.rb:300:in `method_missing'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.16.2/lib/chef/resource_reporter.rb:95:in `block in state'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.16.2/lib/chef/resource_reporter.rb:94:in `each'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.16.2/lib/chef/resource_reporter.rb:94:in `inject'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.16.2/lib/chef/resource_reporter.rb:94:in `state'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.16.2/lib/chef/resource_reporter.rb:65:in `for_json'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.16.2/lib/chef/resource_reporter.rb:273:in `block in prepare_run_data'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.16.2/lib/chef/resource_reporter.rb:272:in `map'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.16.2/lib/chef/resource_reporter.rb:272:in `prepare_run_data'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.16.2/lib/chef/resource_reporter.rb:229:in `post_reporting_data'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.16.2/lib/chef/resource_reporter.rb:214:in `run_completed'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.16.2/lib/chef/event_dispatch/dispatcher.rb:29:in `block in run_completed'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.16.2/lib/chef/event_dispatch/dispatcher.rb:29:in `each'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.16.2/lib/chef/event_dispatch/dispatcher.rb:29:in `run_completed'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.16.2/lib/chef/client.rb:438:in `do_run'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.16.2/lib/chef/client.rb:213:in `block in run'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.16.2/lib/chef/client.rb:207:in `fork'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.16.2/lib/chef/client.rb:207:in `run'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.16.2/lib/chef/application.rb:236:in `run_chef_client'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.16.2/lib/chef/application/client.rb:338:in `block in run_application'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.16.2/lib/chef/application/client.rb:327:in `loop'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.16.2/lib/chef/application/client.rb:327:in `run_application'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.16.2/lib/chef/application.rb:55:in `run'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.16.2/bin/chef-client:26:in `<top (required)>'
/usr/bin/chef-client:23:in `load'
/usr/bin/chef-client:23:in `<main>'
```

**Comment by jtimberman** [comment]
Hi @andrewelizondo,

Can you provide more information about this error? What is the context that you're seeing it? Is this from a custom cookbook? What is the chef-client commandline you used? Can you show more output from the Chef run?

Thanks

**Comment by andrewelizondo** [comment]
@jtimberman 
Sorry! That was my attempt to remember to create the issue with the error while I still had it but I forgot to provide all of the context behind it.

Now to the meat of the explanation. The chef_server_ingredient resource attempts to report using :installed [here](https://github.com/opscode-cookbooks/chef-server-ingredient/blob/master/libraries/chef_server_ingredients_resource.rb#L8) but there is no method inherited or defined so that it can correctly be kicked up to Chef reporting. Because of which the stacktrace only appears at the end of the chef client run after you've already received a run successful message. Commenting out that line has worked for me and I've been using that as a work around in my fork of this.

**Comment by jtimberman** [comment]
I'll follow up next week.

**Comment by jtimberman** [comment]
Can you paste the output of the chef run?

**Comment by jtimberman** [comment]
@andrewelizondo Try with the cookbook off master, this should be resolved in a642ff2
